`test_compression` fails on the latest Fedora Rawhide

Hey!

A while ago I got a notification that the `botan2` package fails to build on Fedora Rawhide due to failing `make check`, and while trying to finally address it I noticed that it is reproducible with the latest `master` branch as well:

```
$ podman run -it --rm fedora:rawhide
# dnf install -y git dnf-plugins-core
# dnf builddep -y botan2
# git clone https://github.com/randombit/botan --depth=1
# cd botan
# ./configure.py --enable-modules=bzip2,zlib
# make -j $(nproc)
# make check |& tee log.txt
...
ZFEC encoding/decoding ran 10449 tests in 6.43 msec all ok
Tests complete ran 2938384 tests in 22.04 sec 3 tests failed (in compression_tests)
ERROR:root:Error running ./botan-test --data-dir=./src/tests/data
make: *** [Makefile:65: check] Error 1
# grep "Failure" log.txt
Failure 1: deflate compression Level 9 compresses text at least as well as level 1 unexpected result 443 < 444 (at src/tests/test_compression.cpp:164)
Failure 1: gzip compression Level 9 compresses text at least as well as level 1 unexpected result 461 < 462 (at src/tests/test_compression.cpp:164)
Failure 1: zlib compression Level 9 compresses text at least as well as level 1 unexpected result 449 < 450 (at src/tests/test_compression.cpp:164)
```

--- a/src/tests/test_compression.cpp
+++ b/src/tests/test_compression.cpp
@@ -99,11 +99,14 @@ class Compression_Tests final : public Test
                result.test_gte("Empty input L1 compresses to non-empty output", c1_e, 1);
                result.test_gte("Empty input L9 compresses to non-empty output", c9_e, 1);
 
-               result.test_gte("Level 9 compresses empty at least as well as level 1", c1_e, c9_e);
-               result.test_gte("Level 9 compresses zeros at least as well as level 1", c1_z, c9_z);
-               result.test_gte("Level 9 compresses random at least as well as level 1", c1_r, c9_r);
-               result.test_gte("Level 9 compresses text at least as well as level 1", c1_t, c9_t);
-               result.test_gte("Level 9 compresses short text at least as well as level 1", c1_s, c9_s);
+               // We assume that Level 9 is better than Level 1, but this is not
+               // guaranteed (see GitHub #3896). Hence, we assert that level 9
+               // it is at most 10% worse than level 1.
+               result.test_gte("Level 9 compresses empty at least as well as level 1", c1_e + (c1_e / 10), c9_e);
+               result.test_gte("Level 9 compresses zeros at least as well as level 1", c1_z + (c1_z / 10), c9_z);
+               result.test_gte("Level 9 compresses random at least as well as level 1", c1_r + (c1_r / 10), c9_r);
+               result.test_gte("Level 9 compresses text at least as well as level 1", c1_t + (c1_t / 10), c9_t);
+               result.test_gte("Level 9 compresses short text at least as well as level 1", c1_s + (c1_s / 10), c9_s);
 
                result.test_lt("Zeros compresses much better than text", c1_z / 8, c1_t);
                result.test_lt("Text compresses much better than random", c1_t / 2, c1_r);

